File: java/programmers/1단계/src/정수제곱근판별.java
public class 정수제곱근판별 {
    public long solution(long n) {
        if (Math.pow((int)Math.sqrt(n), 2) == n) {
            return (long) Math.pow(Math.sqrt(n) + 1, 2);
        }
        return -1;
    }
    public static void main(String[] args) {
        정수제곱근판별 sol = new 정수제곱근판별();
        long n = 121;
        System.out.println(sol.solution(n));
    }
}

File: python/python_Programmers/1단계/정수제곱근판별.py
import math
# 내 풀이
def solution(n):
    answer = 0
    if math.sqrt(n) == int(math.sqrt(n)):
        return int(math.sqrt(n) + 1) **2
    else:
        return -1

# 모범 풀이
def solution2(n):
    sqrt = n ** (1/2)

    if sqrt % 1 == 0:
        return (sqrt + 1) ** 2
    return -1

n = 3
print(solution(n))
print(solution2(n))